src/jekyll/zeppelin.md

```diff
@@ -0,0 +1,69 @@
++ The `Spark deployment modes`: Can be `client` or `cluster`.
+
+Only three combinations of these options are supported:
+
+1. `in-cluster` with `spark-client` mode.
+2. `in-cluster` with `spark-cluster` mode.
+3. `out-cluster` with `spark-cluster` mode.
+
+For now, to be able to test these combinations, you need to build specific branches (see hereafter) or to use third-party Helm charts or Docker images. The needed branches and related PR are listed here:
+
+1. In-cluster client mode [see pull request #456](https://github.com/apache-spark-on-k8s/spark/pull/456)
+2. Add support to run Spark interpreter on a Kubernetes cluster [see pull request #2637](https://github.com/apache/zeppelin/pull/2637)
+
+## In-Cluster with Spark-Client
+
+![In-Cluster with Spark-Client](/img/zeppelin_in-cluster_spark-client.png "In-Cluster with Spark-Client")
+
+Build a new Zeppelin based on [#456 In-cluster client mode](https://github.com/apache-spark-on-k8s/spark/pull/456).
+
+Once done, deploy that new build in a Kubernetes Pod with the following interpreter settings:
+
++ `spark.app.name`: Do not set any name, Zeppelin with pick one for you.
++ `spark.master`: k8s://https://kubernetes:443
++ `spark.submit.deployMode`: client
++ `spark.kubernetes.driver.pod.name`: The name of the pod where your Zeppelin instance is running.
++ Other spark.k8s properties you need to make your spark working (see [Running Spark on Kubernetes](./running-on-kubernetes.html)) such as `spark.kubernetes.initcontainer.docker.image`, `spark.kubernetes.driver.docker.image`, `spark.kubernetes.executor.docker.image`...
+
+## In-Cluster with Spark-Cluster
+
+![In-Cluster with Spark-Cluster](/img/zeppelin_in-cluster_spark-cluster.png "In-Cluster with Spark-Cluster")
+
+Build a new Zepplin  based on [#2637 Spark interpreter on a Kubernetes](https://github.com/apache/zeppelin/pull/2637).
+
+Once done, deploy that new build in a Kubernetes Pod with the following interpreter settings:
+
++ `spark.app.name`: Do not set any name, Zeppelin with pick one for you.
++ `spark.master`: k8s://https://kubernetes:443
++ `spark.submit.deployMode`: cluster
++ `spark.kubernetes.driver.pod.name`: Do not set this property.
++ Other spark.k8s properties you need to make your spark working (see [Running Spark on Kubernetes](./running-on-kubernetes.html)) such as `spark.kubernetes.initcontainer.docker.image`, `spark.kubernetes.driver.docker.image`, `spark.kubernetes.executor.docker.image`...
+
+## Out-Cluster with Spark-Cluster
+
+![Out-Cluster with Spark-Client](/img/zeppelin_out-cluster_spark-cluster.png "Out-Cluster with Spark-Client")
+
+Build a new Spark and their associated docker images based on [#2637 Spark interpreter on a Kubernetes](https://github.com/apache/zeppelin/pull/2637).
+
+Once done, any vanilla Apache Zeppelin deployed in a Kubernetes Pod (your can use a Helm chart for this) will work out-of-the box with the following interpreter settings:
```

Comment:
does this helm chart for this (use a different image for a newer Zeppelin though)
https://github.com/kubernetes/charts/blob/master/stable/spark/templates/spark-zeppelin-deployment.yaml

shall we link it?

Reply:
I have added a section at the end "how to test" and linked to the chart.